analytics/TPO2-Rudin/data_tools/config_ASHB2.txt
; Main configuration file for Columbia-GE Fedex project software
; Module-specific sections override the default section, and command-line
; flags override both
; Leading and trailing spaces stripped from keys and values
; Refer to another key's value with '%(key)s'
; Leave value blank for None value
; Contact a developer if you need literal '%' characters in your value

;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;; Default Settings ;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;

[DEFAULT]

running_at = ccls

;;;; Non-relative locations ;;;;

; root of installation.  Should have subdirs feeders, control, data, etc.
base_dir = C:/Rudin
;base_dir = C:/Users/agagneja/Documents/Rudin
;base_dir = K:/new erc/ConEd Institute/FEDEX/App ;for tiger


; base directory for web-accessible directories
web_base_dir = %(base_dir)s/data

; program locations for use elsewhere
perl_prog = c:/Perl/bin/perl.exe
;perl_prog = /usr/bin/perl
python_prog = C:/Python27/python.exe
;python_prog = N:/Python27/python.exe ;for tiger
;python_prog = /usr/bin/python                          ; maxwell
;python_prog = /usr/local/ActivePython-2.4/bin/python   ; power
;python_prog = /opt/gauss/ActivePython/bin/python    ; franklin


;;;; other locations ;;;;

; default destination for temporary files
temp_dir = %(base_dir)s/Temp

; source of static data and default dest of new models
data_dir = %(base_dir)s/data

; location of static data sources
data_sources_dir = %(base_dir)s/Data-Sources


; directory for control scripts
control_dir = %(base_dir)s/control

; default destination for logs
log_dir = %(base_dir)s/log


;;;; commands ;;;;

ml_dir = %(base_dir)s/ML

;directory for Support Vector Machine programs
svm_dir = %(ml_dir)s/libsvm-3.14
;Support Vector Machine format conversion program
svm_uci = %(svm_dir)s/uci2svml_reg.pl
;Support Vector Machine data normalization program
svm_scale = %(svm_dir)s/windows/svm-scale
;Support Vector Machine prediction program
svm_predict = %(svm_dir)s/svm-rank
;Support Vector Machine training program
svm_train = %(svm_dir)s/svm-train
;Support Vector Machine weight extraction pogram
svm_weight = %(svm_dir)s/svm-weight


;;;; logging configuration ;;;;

; if non-zero, calling app has already configured logger
logger_configured = 0

; $ will be replaced with % before assignment in application code
log_fmt = $(levelname)5s $(module)7s $(asctime)s: $(message)s

; top level threshold
log_level = INFO

; set true to add stderr as a log device
log_console = 1

; Severity threshold for console logging
; Choose from 'CRITICAL', 'ERROR', 'WARN', 'WARNING', 
;   'INFO', 'DEBUG', 'NOTSET'
log_console_level = DEBUG

; message format for console
log_console_msg_fmt = %(log_fmt)s

; set true to add a rotating file as a log device
log_file = 1

; severity threshold for file logging
; Choose from 'CRITICAL', 'ERROR', 'WARN', 'WARNING', 
;   'INFO', 'DEBUG', 'NOTSET'
log_file_level = INFO

; message format for log file
log_file_msg_fmt = %(log_fmt)s

; dir for log file
log_file_dir = %(log_dir)s

; mode for log file, w or a (write or append)
log_file_mode = a

; max size for log file before rotation
log_file_maxbytes = 5000000

; if non-zero, number of old logs to keep
log_file_backup_count = 5

;;;; misc ;;;;

; if non-zero, run in module-defined test mode
test = 0

; if non-zero, run in module-defined debug mode
debug = 0

; common settings for all buildings

; database driver to use
db_driver = {SQL Native Client}

; forecast granularity in minutes; this value is also used to discard input
; data to match the prediction granularity needed
forecast_granularity = 15

; forecast length in hours
forecast_length = 24

; ask Leon
; db_user = rudin_db_user2
; db_pwd = rud1n2013$

building_ids = Rudin_345Park, Rudin_560Lexington
weather_station_ids = weather_central_park_nyc, weather_central_park_nyc

# number of days of recent data to include for training. If available, data from
# a year ago around the as-of-date is also used
training_radius = 30

;space_temp_tablename_format = RUDINSERVER_FL%s_%s_SPACETEMP

; for debugging
use_weather_forecast = 1


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;;;;;;;;;;   Ashwath's Default   ;;;;;;;;;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

; building floor to build model for
building_floors = 
; floor quadrant to build model for respective to the order of floors
floor_quadrants = 
; floor equipments respective to the order of floors
floor_equipments = 
; zone respective to order of floors
floor_zones=

;occupany prediction
occupancy_prediction_table_name = 

predict_occupancy = 

;space temperature prediction
spt_prediction_table_name = 
spt_prediction_table_floor_designator = 
spt_equality_constraint_list = 
spt_prediction_has_quadrant = 

; space temperature
spt_table_name = 


; supply air temperature
sat_table_name = 
sat_equality_constraint_list = 

; return air temperature
rat_table_name =
rat_equality_constraint_list = 

; supply air temperature set point
satS_table_name = 
satS_equality_constraint_list = 

; steam prediction
steam_prediction_table_name = 

;steam actual
steam_table_name

; electricity prediction
elec_prediction_table_name = 

; rampdown
ramp_down_table_name = 

; startup
startup_table_name = 
startup_equality_constraint_list = 

; space temperature trajectory
spt_trajectory_table_name = 
spt_trajectory_table_floor_designator =
spt_trajectory_sat_or_rat =

SERVER = anderson.ldeo.columbia.edu
;192.168.99.12 
;db.aboulanger.com
UID = tpo
PWD = tpo
DB =

; weather db table names
weather_DB = 
weather_obs_table_name = 
weather_hour_table_name = 


; output tables

;previous names
output_startup = startup_time_rec
output_rampdown = rampdown_time_rec

output_space_temp_trajectory = spaceTemperature_trajectory


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;;;;;;;;;;         end           ;;;;;;;;;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;



[weather_central_park_nyc]

; weather related
weather_db = Weather
weather_db_server = anderson.ldeo.columbia.edu
; observed weather data table 
weather_table = Observations_History
weather_forecast_table = Hourly_Forecast

weather_db_user = rudin_db_reader
weather_db_pwd = rudin_20!3$



;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;; Building Configs ;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;


;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;;; Ashwath's Code ;;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;

[345_Park]

DB = 345
weather_DB = Weather

; building floor to build model for
building_floors = 2, 5, 13, 18, 20, 24, 32, 38, 40
; floor quadrant to build model for respective to the order of floors
floor_quadrants = CNW, CSE, CSW, CSE, CSW, CSE, CNW, CNW, CNE

; floor equipments respective to the order of floors
floor_equipments = 010, 012, 009, 011, 004, 006, 004, 003, 005

;occupancy prediction
occupancy_prediction_table_name = [345---------000TPOFORPEOBUI001---001];[345---------000TPOFOROCCCON001---001]

predict_occupancy = 1

;space temperature prediction
spt_prediction_table_name = [345---------000TPOFORTEMSPA001---001];Space_Temp_Prediction
spt_prediction_table_floor_designator = Floor
spt_prediction_has_quadrants = 1

; space temperature
spt_table_name = [345---------001BMSHVATEMSPA---VAL001]
spt_equality_constraint_list = FLOOR, QUADRANT

; supply air temperature
sat_table_name = [345---------001BMSHVAFANSAT---VAL001]
sat_equality_constraint_list = EQUIPMENT_NO

; supply air temperature set point
satS_table_name = [345---------001BMSHVAFANSAT---SPV001]
satS_equality_constraint_list = EQUIPMENT_NO

; return air temperature
rat_table_name = [345---------001BMSHVAFANRAT---VAL001]
rat_equality_constraint_list = EQUIPMENT_NO

; steam prediction
has_steam_supply = 1
steam_prediction_table_name = [345---------000TPOFORSTECON001---001]

; electricity prediction
elec_prediction_table_name = [345---------000TPOFORELECON001---001]

; rampdown
ramp_down_table_name = Ramp_Down_Time_DS

; startup
startup_table_name = [[345---------001BMSCONSOLIDATEDFANS---VAL001]]]#[345---------001BMSCONSOLIDATEDFANS---VAL001]
startup_equality_constraint_list = EQUIPMENT_NO


; space temperature trajectory
;spt_trajectory_table_name = spaceTemperature_trajectory
spt_trajectory_table_name = [345---------000TPONOWTEMSPA001---001]
spt_trajectory_table_floor_designator = Floor
spt_trajectory_sat_or_rat = sat,satS,rat
spt_trajectory_param_savepoint = C:\Rudin\spaceTemperature_trajectory\params\345_Park

;steam
steam_table_name = [345---------001BMSSTEMET------VAL001]

; weather db table names
weather_obs_table_name = Observations_History
weather_hour_table_name = Hourly_Forecast

; output table names
output_startup = [345---------000TPOOPTTEMUPT001---001]
output_rampdown = [345---------000TPOOPTTEMRDW001---001]
output_space_temp_trajectory = [345---------000TPONOWTEMSPA001---001]

building_open_hour = 7
building_close_hour = 19

[560_Lex]

DB = 560
weather_DB = Weather

; building floor to build model for
building_floors = 11, 12, 15, 20, 21, 22, 2, 5, 09
; floor quadrant to build model for respective to the order of floors

;occupancy prediction
occupancy_prediction_table_name =[560---------000TPOFORPEOBUI001---001];[560---------000TPOFOROCCCON001---001]

predict_occupancy = 1

;space temperature prediction
spt_prediction_table_name = [560---------000TPOFORTEMSPA001---001]
spt_prediction_table_floor_designator = Floor
spt_prediction_has_quadrants = 0

; space temperature
spt_table_name = [560---------002BMSHVATEMSPA---VAL001]
spt_equality_constraint_list = FLOOR

sat_table_name = [560---------002BMSHVAFANSAT---VAL001]
; supply air temperature
sat_equality_constraint_list = FLOOR

; return air temperature
rat_table_name = [560---------002BMSHVAFANRAT---VAL001]
rat_equality_constraint_list = FLOOR

; electricity prediction
elec_prediction_table_name = [560---------000TPOFORELECON001---001]

; rampdown
ramp_down_table_name = Ramp_Down_Time_DS

; startup
startup_table_name = [560---------002BMSHVAFAN------VAL001]
startup_equality_constraint_list = FLOOR

; space temperature trajectory
;spt_trajectory_table_name = spaceTemperature_trajectory
spt_trajectory_table_name = [560---------000TPONOWTEMSPA001---001]
spt_trajectory_table_floor_designator = FLOOR
spt_trajectory_sat_or_rat = rat, sat
spt_trajectory_param_savepoint = C:\Rudin\spaceTemperature_trajectory\params\560_Lex

;steam
has_steam_supply = 0
steam_table_name = 

; weather db table names
weather_obs_table_name = Observations_History
weather_hour_table_name = Hourly_Forecast

; output table names
output_startup = [560---------000TPOOPTTEMUPT001---001]
output_rampdown = [560---------000TPOOPTTEMRDW001---001]
output_space_temp_trajectory = [560---------000TPONOWTEMSPA001---001]

building_open_hour = 8
building_close_hour = 19

[40E52]


DB= 4E5
building_db_server = anderson.ldeo.columbia.edu

db_user = rudin_db_user2
db_pwd = rud1n2013$

bms =

; building floor to build model for
building_floors =2, 4, 7, 13, 15, 17#, 21, 21, 24, 24 2, 2, 4, 4, 7, 7, 13, 13, 15, 15, 17, 17#, 21, 21, 24, 24

;How to generalize quadrants?


; floor quadrant to build model for
floor_quadrants = CNO, CNO, CNO, CNO, CSO, CNO#, CNO, CSO, CNO, CSO CNO, CSO, CNO, CSO, CNO, CSO, CNO, CSO, CNO, CSO, CNO, CSO#, CNO, CSO, CNO, CSO

building_open_hour = 7
building_close_hour = 19

weather_station_id = weather_central_park_nyc

; results database
results_db = 4E5
results_db_server = anderson.ldeo.columbia.edu

results_db_user = rudin_db_user2
results_db_pwd =  rud1n2013$

# space temperature predictions

spt_prediction_table_name = [4E5---------000TPOFORTEMSPA001---001]
spt_prediction_table_floor_designator = Floor
spt_prediction_has_quadrants = 1

# space temperature
spt_table_name = [4E5---------003BMSHVATEMSPA---VAL001]
spt_equality_constraint_list = FLOOR, QUADRANT

#supply air temperature
sat_table_name = [4E5---------003BMSHVASFASAT---VAL001]
sat_equality_constraint_list = FLOOR

# return air temperature
rat_table_name = [4E5---------003BMSHVARFARAT---VAL001]
rat_equality_constraint_list = FLOOR

# steam
has_steam_supply = 0 
steam_table_name = 

# electric
elec_table_name = [4E5---------003BMSELEMET------VAL001]

# eletricity prediction
elec_prediction_table_name = [4E5---------000TPOFORELECON001---001]


# output table names
output_startup = [4E5---------000TPOOPTTEMUPT001---001]
output_rampdown = [4E5---------000TPOOPTTEMRDW001---001]
output_space_temp_trajectory = [4E5---------000TPONOWTEMSPA001---001]
# rampdown
ramp_down_table_name = Ramp_Down_Time_DS

# startup
startup_table_name = [4E5---------003BMSHVARFA------VAL001]
startup_equality_constraint_list = FLOOR


# space temperature trajectory
spt_trajectory_sat_or_rat = sat,rat
spt_trajectory_table_name = 
spt_trajectory_table_floor_designator = FLOOR
spt_trajectory_param_savepoint = C:\Rudin\spaceTemperature_trajectory\params\40E52

predict_occupancy = 0


[1BP]

DB= 1BP
SERVER = anderson.ldeo.columbia.edu

UID= rudin_db_user2
PWD= rud1n2013$

bms =

; building floor to build model for
building_floors = 02, 04, 06, 07, 09, 22, 27, 29, 33, 34

; floor quadrant to build model for
floor_quadrants = CEA, CWE, CWE, CEA, CWE, CEA, CEA, CEA, CEA, CWE

floor_zones = Z04,Z04,Z04,Z04,Z04,Z03,Z03,Z03,Z03,Z03,Z01,Z02,Z03,Z04,Z05,Z06,Z07,Z08

holiday_table = 

building_open_hour = 7
building_close_hour = 19

weather_station_id = weather_central_park_nyc

; results database
results_db = 1BP
results_db_server = anderson.ldeo.columbia.edu

results_db_user = rudin_db_user2
results_db_pwd =  rud1n2013$

# space temperature prediction

spt_prediction_table_name = [1BP---------000TPOFORTEMSPA001---001]
spt_prediction_table_floor_designator = Floor
spt_prediction_has_quadrants = 1

# space temperature
spt_table_name = [1BP---------004BMSHVATEMSPA---VAL001]
spt_equality_constraint_list = FLOOR,QUADRANT

# sat
sat_table_name = [1BP---------004BMSHVASFASAT---VAL001]
sat_equality_constraint_list = ZONE

# sat Setpoint
satS_table_name = [1BP---------004BMSHVASFASAT---STP001]
satS_equality_constraint_list = ZONE


# steam
has_steam_supply = 0
steam_prediction_table_name = 
steam_table_name=


# electricity predictions
elec_prediction_table_name= [1BP---------000TPOFORELECON001---001]

#rampdown
ramp_down_table_name= Ramp_Down_Time_DS

#startup
startup_table_name= [1BP---------004BMSHVASFA------VAL001]
startup_equality_constraint_list = ZONE

#rat
rat_table_name = [1BP---------004BMSHVARFARAT---VAL001]
rat_equality_constraint_list = ZONE

#occupancy prediction
predict_occupancy = 0
occupancy_prediction_table = [1BP---------000TPOFORPEOBUI001---001]

#space temp trajectory
spt_trajectory_table = [1BP---------000TPONOWTEMSPA001---001]
spt_trajectory_table_floor_designator = FLOOR
spt_trajectory_sat_or_rat = sat,satS,rat
spt_trajectory_param_savepoint = C:\Rudin\spaceTemperature_trajectory\params\1BP

## outputs (some redundancies!)
output_startup=[1BP---------000TPOOPTTEMUPT001---001]
output_rampdown=[1BP---------000TPOOPTTEMRDW001---001]
output_space_temp_trajectory=[1BP---------000TPONOWTEMSPA001---001]





analytics/TPO2-Rudin/config.py
david = dict(default={
    "base_dir": "/home/davidkarapetyan/Documents/workspace/Adirondack/analytics",
    "python_prog": "/home/davidkarapetyan/anaconda3/bin/python",
    "data_sources": "/data",
    "log_dir": "/var/log"
}, weather={
    "wund_url": "http://api.wunderground.com/api/bab4ba5bcbc2dbec/",
    "table": "/data/weather.h5",
    "history": "history",
    "forecast": "forecast",
    "city": "New_York",
    "state": "NY",
    "cap": 9
}, sampling={
    "forecast_granularity": 15,
    "forecast_length": 24
}, svm={
    "gamma": 1,
    "C": 1,
    "cache_size": 10000,
    "kernel": "rbf",
    "max_iter": -1,
    "prpobability": "False",
    "random_state": "None",
    "shrinking": "True",
    "tol": "0.001",
    "verbose": "False"
}, sarima={
    "order": "(1, 1, 0)",
    "seasonal_order": "(1,1,3)",
    "enforce_stationarity": "False"
})

ashish = None
master = None
